src/sim_engine/mod.rs

```diff
@@ -0,0 +1 @@
+pub mod engine;
```

Comment:
I think you can also re-export stuff here, so in main.rs it could still say `use sim_engine::SimEngine` if desired.